WMF Framework/Event Platform Client/EventPlatformClientLibrary.swift

```diff
@@ -0,0 +1,605 @@
+            } // end if there's a config
+        } // end if there's no cached determination
+        return cache[stream]!
+    }
+
+    /**
+     * Log an event to a stream
+     * - Parameters:
+     *      - stream: Name of the event stream to send the event to
+     *      - schema: Name and version of schema that the instrumentation conforms to
+     *      - data: A dictionary of event data, appropriate for schema version
+     *      - domain: An optional domain to include for the even
+     *
+     * Regarding `schema`, the instrumentation needs to specify which schema (and specifically which
+     * version of that schema) it conforms to. Analytics schemas can be found in the jsonschema directory of
+     * [secondary repo](https://gerrit.wikimedia.org/r/plugins/gitiles/schemas/event/secondary/)
+     *
+     * As an example, if instrumenting client-side error logging, a possible `schema` would be
+     * `/mediawiki/client/error/1.0.0`. For the most part, the `schema` will start with
+     * `/analytics`, since there's where analytics-related schemas are collected.
+     *
+     * Regarding `domain`: this is *optional* and should be used when event needs to be attrributed to a
+     * particular wiki (Wikidata, Wikimedia Commons, a specific edition of Wikipedia, etc.). If the language is
+     * NOT relevant in the context, `domain` can be safely omitted. Using "domain" rather than "language"
+     * is consistent with the other platforms and allows for the possibility of setting a non-Wikipedia domain
+     * like "commons.wikimedia.org" and "wikidata.org" for multimedia/metadata-related in-app analytics.
+     *
+     * Cases where instrumentation would set a `domain`:
+     * - reading or editing an article
+     * - managing watchlist
+     * - interacting with feed
+     * - searching
+     *
+     * Cases where it might not be necessary for instrumentation to set a `domain`:
+     * - changing settings
+     * - managing reading lists
+     * - navigating map of nearby articles
+     * - multi-lingual features like Suggested Edits
+     * - marking session start/end; in which case schema and `data` should have a `languages` field
+     *   where user's list of languages can be stored, although it might make sense to set it to the domain
+     *   associated with the user's 1st preferred language – in which case use
+     *   `MWKLanguageLinkController.sharedInstance().appLanguage?.siteURL()!.host!`
+     */
+    public func log(stream: String, schema: String, data: [String: Any], domain: String? = nil) -> Void {
+        if !ENABLED {
+            return
+        }
+        var meta: [String: String]
+        if data.keys.contains("meta") {
+            meta = data["meta"]! as! [String: String]
+        } else {
+            meta = [String: String]()
+        }
+        if domain != nil {
+            meta["domain"] = domain!
+        }
+        var data = data
+        data["meta"] = meta
```

Comment:
Suggestion to help avoid the `!`

```
        guard enabled else {
            return
        }
        
        var meta: [String: String] = data["meta"] as? [String: String] ?? [:]

        if let domain = domain {
            meta["domain"] = domain
        }
        
        var data = data
        data["meta"] = meta
```

Reply:
Oh that's super cool, thanks! I really like that and it reminds me of the `var meta = eventData.meta || {}` syntax from the JavaScript client